{"name": "test_add_member_fail[user2-11528872802-手机已存在于通讯录，无法添加]", "status": "broken", "statusDetails": {"message": "AttributeError: 'NoneType' object has no attribute 'click'", "trace": "self = <testcases.test_addMember.TestAddMember object at 0x000002637AA436A0>\nuser = 'user2', phone = '11528872802', expect = '手机已存在于通讯录，无法添加'\n\n    @pytest.mark.parametrize(\"user,phone,expect\", test_data['test_add_member_fail'])\n    def test_add_member_fail(self, user, phone, expect):\n>       fail_text = self.addMember = self.app.main().go_to_address().go_to_add_member().add_member_fail(user, phone)\n\ntestcases\\test_addMember.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npage\\main_page.py:14: in go_to_address\n    self.steps(os.getcwd() + \"/pageYaml/main_page.yaml\")\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <page.main_page.Main object at 0x000002637AB13C88>\npath = 'E:\\\\document\\\\code\\\\pythonCode\\\\appium_ui_testFramework/pageYaml/main_page.yaml'\n\n    def steps(self, path):\n        name = inspect.stack()[1].function\n        with open(path, encoding=\"utf-8\") as f:\n            dat = yaml.safe_load(f)\n            steps = dat[name]\n            for step in steps:\n                raw = json.dumps(step)\n                for key, value in self._param.items():\n                    raw = raw.replace(f'${{{key}}}', value)\n                step = json.loads(raw)\n                if \"action\" in step.keys():\n                    if step[\"action\"] == \"click\":\n>                       self.find(step['by'], step['locator']).click()\nE                       AttributeError: 'NoneType' object has no attribute 'click'\n\npage\\base_page.py:64: AttributeError"}, "attachments": [{"source": "ddc7f6b8-9a62-4149-bfd6-ee806036b737-attachment.png", "type": "image/png"}, {"name": "log", "source": "e46ad081-772c-4edb-ab63-748ffa5fb403-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "user", "value": "'user2'"}, {"name": "phone", "value": "'11528872802'"}, {"name": "expect", "value": "'手机已存在于通讯录，无法添加'"}], "start": 1626593166791, "stop": 1626593174723, "uuid": "8ad81e1a-d83b-4998-8d90-6900ee836c3a", "historyId": "f072b275c56a30d0c23d296fd0a67095", "testCaseId": "b251c483add15ec609094c40b1ea12dc", "fullName": "testcases.test_addMember.TestAddMember#test_add_member_fail", "labels": [{"name": "parentSuite", "value": "testcases"}, {"name": "suite", "value": "test_addMember"}, {"name": "subSuite", "value": "TestAddMember"}, {"name": "host", "value": "z"}, {"name": "thread", "value": "18148-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.test_addMember"}]}